src/java.base/share/classes/jdk/internal/io/JdkConsoleImpl.java

```diff
@@ -164,31 +151,23 @@ public void run() {
         shutdownHookInstalled = true;
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public char[] readPassword() {
         return readPassword("");
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public void flush() {
         pw.flush();
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public Charset charset() {
-        assert CHARSET != null : "charset() should not return null";
-        return CHARSET;
+        assert charset != null : "charset() should not return null";
```

Comment:
Maybe we could add a `Objects.requireNonNull(charset)` in the constructor of this `JdkConsoleImpl` and remove this `assert`, since `charset` is `final`?

Reply:
Good point. Replaced `assert` with `Objects.requireNonNull(charset)`